allennlp/data/fields/text_field.py

```diff
@@ -83,11 +106,20 @@ def get_padding_lengths(self) -> Dict[str, int]:
                 token_lengths = [{}]
             # Iterate over the keys and find the maximum token length.
             # It's fine to iterate over the keys of the first token since all tokens have the same keys.
-            for key in token_lengths[0].keys():
+            for key in token_lengths[0]:
                 indexer_lengths[key] = max(x[key] if key in x else 0 for x in token_lengths)
             lengths.append(indexer_lengths)
-        any_indexed_token_key = list(self._indexed_tokens.keys())[0]
-        padding_lengths = {'num_tokens': len(self._indexed_tokens[any_indexed_token_key])}
+
+        # Get the padding lengths for sequence lengths.
+        if len(set(indexer_sequence_lengths.values())) == 1:
+            # This is the default case where all indexers return the same length.
+            padding_lengths = {'num_tokens': list(indexer_sequence_lengths.values())[0]}
+        else:
+            # The indexers return different lengths.
+            padding_lengths = {
+                    'num_tokens' + '-_-' + key: val for key, val in indexer_sequence_lengths.items()
```

Comment:
Could you explain a bit why this `-_-` is necessary? I don't exactly understand... Is it because we are returning a flat dictionary of lengths, but there could be multiple arrays  returned which have the same keys but different values?

Reply:
Since we flatten the dict for all token indexers, then add additional keys, I inserted this here to avoid potentially clobbering keys that get returned from one of the other token indexers.